File: official/docs/csharp/v5/carbon-offset/buy.cs
using System;
using System.Collections.Generic;
using System.Threading.Tasks;
using Newtonsoft.Json;
using EasyPost;
using EasyPost.Models.API;
using EasyPost.Parameters;

namespace EasyPostExamples
{
    public class Examples
    {
        public static async Task Main()
        {
            var client = new EasyPost.Client("EASYPOST_API_KEY");

            Shipment shipment = await client.Shipment.Retrieve("shp_...");

            Rate rate = shipment.LowestRate();

            Parameters.Shipment.Buy parameters = new(rate)
            {
                CarbonOffset = true,
            };

            shipment = await client.Shipment.Buy(shipment.Id, parameters);

            Console.WriteLine(JsonConvert.SerializeObject(shipment, Formatting.Indented));
        }
    }
}

File: official/docs/node/current/shipments/buy.js
const EasyPostClient = require('@easypost/api');

const client = new EasyPostClient('EASYPOST_API_KEY');

(async () => {
  const retrievedShipment = await client.Shipment.retrieve('shp_...');

  const shipment = await client.Shipment.buy(
    retrievedShipment.id,
    retrievedShipment.lowestRate(),
    249.99,
  );

  console.log(shipment);
})();
